File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java
import java.io.*;
import java.util.*;

public class Main {
	static int N, M;
	static int[][] graph;
	static boolean[] visited;
	public static void main(String[] args) throws Exception {
		BufferedReader br=new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st=new StringTokenizer(br.readLine(), " ");
		
		N=Integer.parseInt(st.nextToken());
		M=Integer.parseInt(st.nextToken());
		int V=Integer.parseInt(st.nextToken());
		
		graph=new int[N+1][N+1];
		visited=new boolean[N+1];
		for(int i=0; i<M; i++) {
			st=new StringTokenizer(br.readLine(), " ");
			int from=Integer.parseInt(st.nextToken());
			int  to =Integer.parseInt(st.nextToken());
			graph[from][to]=graph[to][from]=1;
		}
		dfs(V);
		System.out.println();
		visited=new boolean[N+1];
		bfs(V);
	}
	static void dfs(int start) {
		visited[start]=true;
		System.out.print(start+" ");
		for(int j=1; j<=N; j++) {
			if(graph[start][j]!=0 && !visited[j]) {
				dfs(j);
			}
		}
	}
	static void bfs(int start) {
		ArrayDeque<Integer> queue=new ArrayDeque<>();
		visited[start]=true;
		queue.offer(start);
		while(!queue.isEmpty()) {
			start=queue.poll();
			System.out.print(start+" ");
			for(int j=1; j<=N; j++) {
				if(graph[start][j]!=0 && !visited[j]) {
					visited[j]=true;
					queue.offer(j);
				}
			}
		}
	}
}

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.cc
#include <iostream>
#include <vector>
#include <algorithm>
#include <queue>
using namespace std;

static vector<vector<int>> graph;
static vector<bool> visited;
void dfs(int start);
void bfs(int start);
int main()
{
	ios::sync_with_stdio(false);
	cin.tie(NULL);
    cout.tie(NULL);

	int N, M, V;
	cin >> N >> M >> V;

	graph.resize(N + 1);

	for (int i = 0; i < M; i++) {
		int from, to;
		cin >> from >> to;

		graph[from].push_back(to);
		graph[to].push_back(from);
	}

	for (int i = 1; i <= N; i++) {
		sort(graph[i].begin(), graph[i].end());
	}

	visited = vector<bool>(N + 1, false);

	dfs(V);
	cout << "\n";
	visited = vector<bool>(N + 1, false);
	bfs(V);
}
void dfs(int start) {
	visited[start] = true;
	cout << start << " ";
	for (int i = 0; i < graph[start].size(); i++) {
		int next = graph[start][i];
		if (!visited[next]) dfs(next);
	}
}
void bfs(int start) {
	queue<int> q;
	q.push(start);
	visited[start] = true;
	while (!q.empty()) {
		int cur = q.front();
		q.pop();
		cout << cur << " ";
		for (int i = 0; i < graph[cur].size(); i++) {
			int next = graph[cur][i];
			if (!visited[next]) {
				visited[next] = true;
				q.push(next);
			}
		}
	}
}